src/transparency/client.rs

```diff
@@ -14,9 +14,17 @@ use telemetry::sensor::http::RequestBody;
 use super::glue::{BodyPayload, HttpBody, HyperConnect};
 use super::upgrade::{HttpConnect, Http11Upgrade};
 
+use std::{self, fmt};
+
 type HyperClient<C, B> =
     hyper::Client<HyperConnect<C>, BodyPayload<RequestBody<B>>>;
 
+#[derive(Debug)]
+pub enum Error {
+    Http1(hyper::Error),
+    Http2(tower_h2::client::Error),
```

Comment:
This makes it seem like non-version-specific errors will be reported two different ways depending what the underyling version is. I suspect there are very few errors that are really protocol-version-specific, and I also suspect that the protocol-version-specific errors are mostly H2-specific and not very interesting to end users, so it seems like it would be good to find a way to unify these errors in the future.

Reply:
Yea, we definitely should! A problem currently is that metrics (really, tap) requires that we send an HTTP2 Error Code if there is any error at all, which doesn't make sense if the request were HTTP/1 (for reference, in that case, I believe we just report `INTERNAL_ERROR`...).

Reply:
> This makes it seem like non-version-specific errors will be reported two different ways depending what the underyling version is

That's a good point. I think ideally, we'd want to handle cases where the error was caused by, say, an `io::Error` and map that to a separate variant, and restrict the `Http1`/`Http2` variants to errors specific to those protocols. For now, though, I think I'll just add a comment saying that these refer to errors in the HTTP/1 or HTTP/2 _client_ rather than errors specific to those protocols.

Reply:
> A problem currently is that metrics (really, tap) requires that we send an HTTP2 Error Code if there is any error at all, which doesn't make sense if the request were HTTP/1 (for reference, in that case, I believe we just report INTERNAL_ERROR...)

I'd like to fix this, but my understanding is that it isn't currently prioritized.